examples/coap/src/bin/coapserver-coaphandler.rs

```diff
@@ -0,0 +1,159 @@
+            println!("PRK_out: {:02x?}", prk_out);
+
+            let mut _oscore_secret = responder.edhoc_exporter(0u8, &[], 16).unwrap(); // label is 0
+            println!("OSCORE secret: {:02x?}", _oscore_secret);
+            let mut _oscore_salt = responder.edhoc_exporter(1u8, &[], 8).unwrap(); // label is 1
+            println!("OSCORE salt: {:02x?}", _oscore_salt);
+
+            // context of key update is a test vector from draft-ietf-lake-traces
+            let prk_out_new = responder.edhoc_key_update(&[
+                0xa0, 0x11, 0x58, 0xfd, 0xb8, 0x20, 0x89, 0x0c, 0xd6, 0xbe, 0x16, 0x96, 0x02, 0xb8,
+                0xbc, 0xea,
+            ]);
+            println!("PRK_out after key update: {:02x?}?", prk_out_new);
+
+            _oscore_secret = responder.edhoc_exporter(0u8, &[], 16).unwrap(); // label is 0
+            println!("OSCORE secret after key update: {:02x?}", _oscore_secret);
+            _oscore_salt = responder.edhoc_exporter(1u8, &[], 8).unwrap(); // label is 1
+            println!("OSCORE salt after key update: {:02x?}", _oscore_salt);
+
+            EdhocResponse::Message3Processed
+        }
+    }
+    fn estimate_length(&mut self, _: &Self::RequestData) -> usize {
+        200
+    }
+    fn build_response(
+        &mut self,
+        response: &mut impl coap_message::MutableWritableMessage,
+        req: Self::RequestData,
+    ) {
+        response.set_code(coap_numbers::code::CHANGED.try_into().ok().unwrap());
+        match req {
+            EdhocResponse::OkSend2 { c_r } => {
+                let responder = self.connection_by_c_r(c_r).unwrap();
+                let message_2 = responder.prepare_message_2(c_r).unwrap();
+                response.set_payload(&message_2.content[..message_2.len]);
+            }
+            EdhocResponse::Message3Processed => (), // "send empty ack back"?
+        };
+    }
+}
+
+fn build_handler() -> impl coap_handler::Handler {
+    use coap_handler_implementations::{HandlerBuilder, ReportingHandlerBuilder};
+
+    let mut edhoc: EdhocHandler = Default::default();
+
+    coap_handler_implementations::new_dispatcher()
+        .at_with_attributes(&[".well-known", "edhoc"], &[], edhoc)
+        .with_wkc()
+}
+
+fn main_on_stack<S: UdpFullStack>(stack: &mut S) {
+    let mut sock = stack.socket().expect("Can't create a socket");
+
+    let mut handler = build_handler();
+
+    stack.bind(&mut sock, 5683).expect("Can't bind to port");
+
+    loop {
+        match embedded_nal_minimal_coapserver::poll(stack, &mut sock, &mut handler) {
+            Err(embedded_nal::nb::Error::WouldBlock) => {
+                // See <https://github.com/rust-embedded-community/embedded-nal/issues/47>
+                std::thread::sleep(std::time::Duration::from_millis(50));
+            }
+            e => e.expect("UDP error during send/receive"),
+        }
+    }
+}
+
+fn main() {
+    let mut stack = std_embedded_nal::Stack::default();
```

Comment:
Do you think it would make sense to rename this to `network_stack`? At first I was thinking it was the program stack 🙃.